plugins/cpp_metrics/service/src/cppmetricsservice.cpp

```diff
@@ -16,6 +16,35 @@ CppMetricsServiceHandler::CppMetricsServiceHandler(
 {
 }
 
+double CppMetricsServiceHandler::getCppMetricsForAstNode(
+  const core::AstNodeId& astNodeId_,
+  const CppMetricsType::type metrics_)
+{
+  return astNodeMetrics(astNodeId_, metrics_);
+}
+
+double CppMetricsServiceHandler::astNodeMetrics(
+  const ::cc::service::core::AstNodeId& astNodeId_,
+  const CppMetricsType::type type_)
+{
+  _transaction([&, this](){
+    typedef odb::result<model::CppAstNode> CppAstNodeResult;
+    typedef odb::query<model::CppAstNode> CppAstNodeQuery;
+    typedef odb::result<model::CppAstNodeMetrics> CppAstNodeMetricsResult;
+    typedef odb::query<model::CppAstNodeMetrics> CppAstNodeMetricsQuery;
+
+    auto nodeMetric = _db->query<model::CppAstNodeMetrics>(
+      CppAstNodeMetricsQuery::astNodeId == std::stoull(astNodeId_) &&
+      CppAstNodeMetricsQuery::type ==
+        static_cast<model::CppAstNodeMetrics::Type>(type_));
+
+    if (nodeMetric.empty())
+      return -1.0;
```

Comment:
-1.0 could be a valid value for some metrics, do not return it as an extremal value.

Reply:
I changed it to `-DBL_MAX` as I read it on Stackoverflow. :)

Reply:
I still think no double value should be used here as an extremal value. Either `null` or e.g. a 404 status code should be returned.

(Also, in C++ code, the `std::numeric_limits<double>::max()` const function should be preferred over the C macro `DBL_MAX`.)

Reply:
Changed it to throwing an error, as we discussed today.